app/src/main/java/com/google/android/ground/ui/util/PolygonUtil.kt

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright 2025 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.android.ground.ui.util
+
+import com.google.android.ground.model.geometry.Coordinates
+import java.util.Locale
+import kotlin.math.abs
+import kotlin.math.cos
+
+// Calculates the area of a polygon using the Shoelace formula
+fun calculateShoelacePolygonArea(coordinates: List<Coordinates>): Double {
+  if (coordinates.size < 3) return 0.0
+
+  val reference = coordinates[0]
+  val points = coordinates.map { toMeters(reference, it) }
+
+  var area = 0.0
+  for (i in points.indices) {
+    val j = (i + 1) % points.size
+    val term = points[i].first * points[j].second - points[j].first * points[i].second
+    area += term
+  }
+
+  val finalArea = abs(area) / 2.0
+
+  return String.format(Locale.US, "%.2f", finalArea).toDouble()
```

Comment:
Why convert to a String then back to a double? Rounding etc should be done in the UI layer, not done during calculation.